lib/translator/read_capacity_10.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "read_capacity_10.h"
+
+#include <cstring>
+
+namespace translator {
+
+StatusCode ReadCapacity10ToNvme(absl::Span<const uint8_t> raw_scsi,
+                                nvme::GenericQueueEntryCmd& identify_ns,
+                                scsi::LunAddress lun) {
+  scsi::ReadCapacity10Command cmd = {};
+  if (!ReadValue(raw_scsi, cmd)) {
+    DebugLog("Malformed ReadCapacity10 Command");
+    return StatusCode::kInvalidInput;
+  };
+
+  if (cmd.control_byte.naca == 1) {
+    DebugLog("Malformed ReadCapacity10 Command");
```

Comment:
Might be useful to print a slightly more detailed/different error message here so it's easier to tell where things are going wrong when debugging